.github/workflows/smoke-tests.yml

```diff
@@ -87,4 +87,18 @@ jobs:
           exit 1
         fi
 
-
+    - name: Port Forward frontend
+      run: |  
+        set -x
+        port_forward() {
+            # get name of frontend pod
+            local frontend_pod=$(kubectl get pods -l=app=frontend -o=jsonpath='{.items[0].metadata.name}')
+            kubectl port-forward $frontend_pod 3000:8080   
+        }
+        port_forward &
+    - name: Ping frontend
+      run: |
+          curl --silent --output /dev/null --show-error --fail localhost:3000 
+    - name: UI tests
+      run: |
+        docker run -t -v $PWD/e2e:/e2e -w /e2e --network=host cypress/included:4.3.0
```

Comment:
If cypress is already packaged as a container, can't we just 'kubectl run' it in the cluster? That way we wouldn't have to worry about port forwarding. We'd just need to find a way to mount the e2e folder

Reply:
I would have to dig in a bit to verify that change on my end.  Any reasons why the port-forwarding should be avoided?

Reply:
- I'm concerned that the port-forward process could stick around between runs, reserving port 3000 and causing issues (although from your last comment I'm not sure if we need to worry about this)
- Kubernetes and Docker have their own abstractions for stuff like networking, and it would be a lot cleaner if we could stick to one platform instead of trying to get them to talk. I'm assuming this test will get more complex over time, so we might appreciate being in the same virtual network as all the other services at some point later.

But overall if the Github Actions docs say it's ok to run background processes and they will be cleaned up for us, I'm not too worries either way

Reply:
As an added precaution, can we not add a step to clean up any process on particular ports if that is a concern?

Reply:
We can definitely add a clean up process if it becomes an issue.  I don't think it's necessary to add at the moment since the actions runner [cleans up orphaned processes](https://github.com/actions/runner/blob/b0a1294ef5751746eb2b4815c4e3e8a6d575833c/src/Runner.Worker/JobExtension.cs#L271) and surfaces it in the [check's logs](https://pipelines.actions.githubusercontent.com/BR7w0sd8lc5VMu0meoJMolOSMdu4pnfw1Nj5iUBFLi2lDnme12/_apis/pipelines/1/runs/440/signedlogcontent/3?urlExpires=2020-04-13T20%3A01%3A35.4466587Z&urlSigningMethod=HMACV1&urlSignature=QPTqk66ghy3gC4C81h5aoGY4OyzWQL2Igu3oO3vqNeo%3D)

The runner's logs also show this:
```
[2020-04-10 22:16:31Z INFO JobExtension] Inspecting process environment variables. PID: 4440 (kubectl)
```